crates/client/block_import/src/metrics.rs

```diff
@@ -70,16 +138,17 @@ impl BlockMetrics {
         };
 
         let total_sync_time = now.duration_since(self.starting_time).as_secs_f64();
-        self.l2_sync_time.set(total_sync_time);
-        self.l2_latest_sync_time.set(latest_sync_time);
-        self.l2_avg_sync_time.set(total_sync_time / (block_header.block_number - self.starting_block) as f64);
 
-        self.l2_block_number.set(block_header.block_number as f64);
-        self.transaction_count.set(f64::from_u64(block_header.transaction_count).unwrap_or(0f64));
-        self.event_count.set(f64::from_u64(block_header.event_count).unwrap_or(0f64));
+        self.l2_sync_time.record(total_sync_time, &[]);
+        self.l2_latest_sync_time.record(latest_sync_time, &[]);
+        self.l2_avg_sync_time.record(total_sync_time / (block_header.block_number - self.starting_block) as f64, &[]);
+
+        self.l2_block_number.record(block_header.block_number as f64, &[]);
+        self.transaction_count.record(f64::from_u64(block_header.transaction_count).unwrap_or(0f64), &[]);
```

Comment:
IMO : 
1. `l2_avg_sync_time` should be a `f64` since `total_sync_time` is a `f64`.
2. `l2_block_number` should be used as an `u64` and is changed accordingly.
3. `transaction_count` should be used as an `u64` and is changed accordingly.

Committed [here](https://github.com/madara-alliance/madara/pull/348/commits/731f7933090a42efa9052d1ab6f3ecfee2ab83c2).